docs/_source/docs/resolvers.rst

```diff
@@ -79,6 +79,37 @@ A resolver to execute any shell command.
 
 Refer to `sceptre-resolver-cmd <https://github.com/Sceptre/sceptre-resolver-cmd/>`_ for documentation.
 
+stack_attr
+~~~~~~~~~~
+
+This resolver lets you reference the values of other fields on the same Stack Config or those
+inherited from StackGroups in which the Stack Config exists. It is useful for avoiding unnecessary
+configuration duplication as well as referencing attributes inherited from the StackGroup Config.
+
+For example, if you need to reference the ``template_bucket_name`` set in the StackGroup Config,
+you can do that with ``!stack_attr``:
+
+.. code-block:: yaml
+
+   template:
+       type: sam
+       path: path/from/my/cwd/template.yaml
+       # template_bucket_name could be set by a resolver in the StackGroup.
+       artifact_bucket_name: !stack_attr template_bucket_name
```

Comment:
I think users would expect that  `artifact_bucket_name: {{ stack_group_config.template_bucket_name }}` would work as well.  if it doesn't then can you please provide some explanation as to why it would not work?

Reply:
does the template handler doc need updating? https://github.com/Sceptre/sceptre-sam-handler#arguments section.. 
```
artifact_bucket_name (string, required): The bucket name where artifacts should be uploaded to on S3 during the packaging process. If your project has a template_bucket_name, you can set this to {{ template_bucket_name }}.
```

Reply:
> I think users would expect that artifact_bucket_name: {{ stack_group_config.template_bucket_name }} would work as well

The problem here is that template_bucket_name (and many other items that might be in stack_group_config) might be resolvers. Especially with the recent set of merges to master, there are a lot more resolvable properties and "inheriting" a resolver is a lot more likely. Remember, the order of resolution within Sceptre is:

1. Gather raw stack group config variables from higher-level stack group config.yaml files. Since the yaml resolver constructs the resolvers, these variables might not be strings but instead might be raw resolvers.
2. (**"Render time"**) Use those stack group config and user variables to _render_ the config file using Jinja into a string. Thus `{{ stack_group_config.template_bucket_name }}` would cast the resolver to string rather than resolve it.
3. Parse the string into yaml, which will instantiate resolvers on the Stack.
4. Use that yaml to create the Stack instance, which assigns those resolvers to resolvable properties
5. (**"Resolve time"**) Then, upon accessing those resolvable properties, those resolvers are resolved into values.

Because of this, you cannot access resolved values at render time (step 2 above). This is the whole reason why I made the `!stack_attr` resolver, because we need to access other stack attributes (which are set at step 4 and resolved at step 4). Simply accessing the pre-resolved values at render time doesn't work. 

> if it doesn't then can you please provide some explanation as to why it would not work?

I hope the above works. However, I was thinking that one thing that hasn't been documented in Sceptre's docs is the "Stack construction lifecycle", similar to what I just wrote above. I'll add a docs page that explains this, since I find it's frequently confused in the conversations I've had with you and other maintainers lately.

> does the template handler doc need updating? https://github.com/Sceptre/sceptre-sam-handler#arguments section.

Yeah, it will. However, I didn't want to update it until this change has been released as a version. Otherwise, I'd be documenting functionality that doesn't exist yet. The extension repos make more sense if they "chase" Sceptre versions rather than attempt to preempt them.

Reply:
> The problem here is that template_bucket_name (and many other items that might be in stack_group_config) might be resolvers.

What if `template_bucket_name` is not a resolver?  I have template_bucket_name set to a static value in my StackGroupConfig (config.yaml) files.  Should I not be able to use {{ stack_group_config.template_bucket_name }} in my stack config files in this case?  I'm just saying that is is documenting for a specific case and you should make it clear why `!stack_attr` needs to be used instead of `{{ stack_group_config.template_bucket_name }}`

Reply:
Great point! I'll make that change.